File: Objective-C/macOS/PrimaryViewController.m
/*
See LICENSE folder for this sample’s licensing information.

Abstract:
A base table view controller to share a table view between subclasses. Allows its subclasses to display available products, invalid identifiers,
 purchases, and restored purchases.
*/

#import "PrimaryViewController.h"

@implementation PrimaryViewController
#pragma mark - Reload UI

/// Reloads the UI with new products, invalid identifiers, or purchases.
-(void)reloadUIWithData:(NSMutableArray *)data {
	self.data = data;
	[self.tableView reloadData];
}

#pragma mark - NSTable​View​Data​Source

- (NSInteger)numberOfRowsInTableView:(NSTableView *)tableView {
	return self.data.count;
}

@end

File: Swift/macOS/PrimaryViewController.swift
/*
See LICENSE folder for this sample’s licensing information.

Abstract:
A base table view controller to share a table view between subclasses. Allows its subclasses to display available products, invalid identifiers,
purchases, and restored purchases.
*/

import Cocoa

class PrimaryViewController: NSViewController, NSTableViewDataSource, NSTableViewDelegate {
	// MARK: - Properties

	/// Data model used by all PrimaryViewController subclasses.
	var data = [Any]()

	/// Table view used by all PrimaryViewController subclasses.
	@IBOutlet weak var tableView: NSTableView!

	// MARK: - NSTable​View​Data​Source

	func numberOfRows(in tableView: NSTableView) -> Int {
		return data.count
	}
}
